src/montecarlo.jl

```diff
@@ -0,0 +1,103 @@
+export MetropolisMonteCarlo,
+    random_uniform_translation!,
+    random_normal_translation!
+
+"""
+    MetropolisMonteCarlo(; <keyword arguments>)
+
+A Monte Carlo simulator for [`System`](@ref) that uses the Metropolis algorithm to sample the configuration space.
+The method of `simulate!` for this simulator accepts an optional keyword argument `log_states::Bool=true` which 
+tells whether to run the loggers or not (for example, during equilibration).
+
+# Arguments
+- `temperature::T`: The temperature of the system.
+- `log_interval::Int`: The interval at which to log the system state.
+- `trial_moves::UP`: A function that performs the trial moves.
+"""
+struct MetropolisMonteCarlo{T, M}
+    temperature::T
+    trial_moves::M
+    trial_args::Dict
+end
+
+function MetropolisMonteCarlo(;temperature::T,
+                                trial_moves::M,
+                                trial_args::Dict=Dict()) where {T, M}
+    return MetropolisMonteCarlo{T, M}(temperature, trial_moves, trial_args)
+end
+
+function simulate!(sys::System{D, G, T},
+                    sim::MetropolisMonteCarlo,
+                    n_steps::Int;
+                    n_threads::Int=Threads.nthreads(),
+                    log_states::Bool=true) where {D, G, T}
+    if dimension(sys.energy_units) == u"𝐋^2 * 𝐌 * 𝐍^-1 * 𝐓^-2"
+        k_b = sys.k * T(Unitful.Na) 
+    else
+        k_b = sys.k
+    end
+
+    neighbors = find_neighbors(sys, sys.neighbor_finder; n_threads=n_threads)
+    E_old = potential_energy(sys, neighbors)
+    for i in 1:n_steps
+        coords_old = copy(sys.coords)
+        sim.trial_moves(sys; sim.trial_args...)  # changes the coordinates of the system
+        neighbors = find_neighbors(sys, sys.neighbor_finder; n_threads=n_threads)
+        E_new = potential_energy(sys, neighbors)
+
+        ΔE = E_new - E_old
+        δ = ΔE / (k_b * sim.temperature)
+        if δ < 0 || rand() < exp(-δ)
+            log_states && run_loggers!(sys, neighbors, i; n_threads=n_threads, success=true,
+                        energy_rate=E_new / (k_b * sim.temperature))
+            E_old = E_new
+        else
+            sys.coords = coords_old
+            log_states && run_loggers!(sys, neighbors, i; n_threads=n_threads, success=false,
+                        energy_rate=E_old / (k_b * sim.temperature))
+        end
+    end
+end
+
+"""
+	random_uniform_translation!(sys::System; scaling=1.0, length_units=unit(sys.coords[1][1]))
```

Comment:
I think `scaling` and `length_units` can be one argument, something like `shift_size`.